cpp/src/Slice/Parser.cpp

```diff
@@ -2208,26 +2209,16 @@ Slice::Container::checkIntroduced(const string& scopedName, ContainedPtr namedTh
         // We've previously introduced the first component to the current scope, check that it has not changed meaning.
         if (it->second->scoped() != namedThing->scoped())
         {
-            // Parameter are in its own scope.
-            if ((dynamic_pointer_cast<Parameter>(it->second) && !dynamic_pointer_cast<Parameter>(namedThing)) ||
-                (!dynamic_pointer_cast<Parameter>(it->second) && dynamic_pointer_cast<Parameter>(namedThing)))
-            {
-                return true;
-            }
-
-            // Data members are in its own scope.
-            if ((dynamic_pointer_cast<DataMember>(it->second) && !dynamic_pointer_cast<DataMember>(namedThing)) ||
-                (!dynamic_pointer_cast<DataMember>(it->second) && dynamic_pointer_cast<DataMember>(namedThing)))
+            // Parameters and data members are in their own scopes.
```

Comment:
I combined the two `if`s from before, and negated them.
Since logically, we really only do something in the 'else' branch of the previous code.